src/codesyncd/validators/tab_validator.ts

```diff
@@ -0,0 +1,63 @@
+/*
+class TabValidator:
+	validateRepoID()
+	validate_yml_file()
+	group_data()
+*/
+
+import { REQUIRED_KEYS_TAB_FILE_YML, TAB_SIZE_LIMIT } from "../../constants";
+import { ITabYML } from "../../interface";
+import { CodeSyncLogger } from "../../logger";
+import { generateSettings } from "../../settings";
+import { readYML } from "../../utils/common";
+import { ConfigUtils } from "../../utils/config_utils";
+import { removeFile } from "../../utils/file_utils";
+
+export class TabValidator {
+
+	settings: any;
+	configJSON: any;
+	constructor() {
+		this.settings = generateSettings();
+        this.configJSON = readYML(this.settings.CONFIG_PATH);
+
+	}
+
+	validateRepo(tabFile: ITabYML, tabData: string, filePath: string, repoId: number) {
+		const config_utils = new ConfigUtils();
+		const repo_path = config_utils.getRepoPathByRepoId(repoId);
+		const configRepo = this.configJSON.repos[repo_path];
```

Comment:
Where are we validating that the Repo Belongs to Logged-in User?

Reply:
Added the following: 

![image](https://github.com/codesyncapp/codesync-vs/assets/15974895/fa24ff1a-c916-469b-a534-c64a690d1461)

where `repoEmail = configRepo.email;`